Брозовский Максим

BMV989 opened the pull request:
@ksamnole

BMV989 committed:
Homework solution

BMV989 committed:
Added approval tests

BMV989 committed:
small fix

BMV989 committed:
another small fix

ksamnole commented on ObjectPrinting/ObjectPrinting.csproj:
```diff
@@ -5,8 +5,45 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="ApprovalTests" Version="7.0.0-beta.3" />
```
Лучше не использовать бета версии

BMV989 replied:
Там трабла, стабильная не работает (валится страшный error, что чего-то нет), поэтому заюзал ласт версию

ksamnole replied:
ok

ksamnole commented on ObjectPrinting/ObjectPrinting.csproj:
```diff
@@ -5,8 +5,45 @@
     <PackageReference Include="NUnit3TestAdapter" Version="4.6.0" />
   </ItemGroup>
+
+  <ItemGroup>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.Demo.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WithCustomSerializationForSpecificType.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WhenSerializingWithCustomCulture.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WhenTrimmingStringProperties.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WhenExcludingSpecificProperty.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WhenChainingSerializationRules.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WhenUsingExtensionMethod.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WithConfiguration.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldDetectCyclicReferences.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WhenSerializingDictionary.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
+    <None Update="Tests\ObjectPrinterAcceptanceTests.PrintToString_ShouldWorkCorrectly_WhenSerializingEnumerable.approved.txt">
+      <DependentUpon>ObjectPrinterAcceptanceTests.cs</DependentUpon>
+    </None>
```
Вот этого не должно быть)

BMV989 replied:
Убрал

ksamnole commented on ObjectPrinting/Tests/ObjectPrinterAcceptanceTests.PrintToString_ShouldDetectCyclicReferences.approved.txt:
```diff
@@ -0,0 +1,17 @@
+﻿Person
+	Id = Guid
```
Точно не помню как в задании, но вроде для Guid тоже должно прописываться значение. Если что поправь меня

BMV989 replied:
Поправил

ksamnole commented on ObjectPrinting/PrintingConfig.cs:
```diff
@@ -1,41 +1,152 @@
+
+    public string PrintToString(TOwner obj) => PrintToString(obj, 0);
+
+    public void AddSerializerForType<TPropType>(Func<TPropType, string> serializer) =>
+        typeSerializers[typeof(TPropType)] = obj => serializer((TPropType)obj);
+
+    public void AddSerializerForProperty(string propertyName, Func<object, string> serializer) =>
+        propertySerializers[propertyName] = serializer;
+
+    public void AddCultureForType<TPropType>(CultureInfo culture) => typeCultures[typeof(TPropType)] = culture;
```
Кажется, что эти методы либо не должны быть публичными, либо должны возвращать PropertyPrinterConfig. Я еще не глядя в код создал твой objectPrinter и попытался интуитивно его использовать, как и определяет fluent api. Но наткнулся на эти методы и не понимал почему добавляя сериализацию - я не могу и дальше задавать правила "принтинга"

ksamnole commented:
В общем всё ок! Поставлю пока 1 балл, как исправишь замечания сразу исправлю на полные баллы

BMV989 committed:
refactoring:
-- removed dependent files
-- now printing GUID
-- changed methods signatures to be more fluent
